discovery-provider/src/solana/solana_program_indexer.py

```diff
@@ -226,6 +226,15 @@ def get_transaction_batches_to_process(self) -> List[List[str]]:
                 intersection_found = True
                 self.msg(f"No transactions found before {last_tx_signature}")
             else:
+                # truncate repeated trailing slots that could be incomplete [4, 3, 2, 2, 2...] -> [2, 1]
```

Comment:
I'm not sure this makes sense - do we want to drop slot 3 here? Doesn't the fact that 4 came through mean 3 is complete? Also, this doesn't seem to work for sparse transactions. If we get one transaction for 4 but don't see 5 for a while, does 4 just sit unindexed for a while too? This is why we were grabbing the slot height all-up since the slots will def change there

Reply:
Chatted offline - this comment just threw me and my understanding was backwards: we actually know that 4 is finalized, so the real overlap problem is on _old_ slots, not new slots, being in different batches. So this code actually truncates off 2 for the next batch.

The only risk here is that one batch could be 100% from one slot, in which case we would stall?

Reply:
yeah that's why i brought up that case in slack gonna add it to the PR description